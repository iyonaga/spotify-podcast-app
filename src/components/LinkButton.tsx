import Link from 'next/link';

interface Props {
  children: React.ReactNode;
  href: string;
  isExternal?: boolean;
}

const LinkButton: React.VFC<Props> = ({
  children,
  href,
  isExternal,
  ...props
}) => {
  const Button = (
    <div className="inline-block">
      <div className="flex justify-center items-center px-[20px] min-w-[160px] h-[40px] text-[14px] font-bold hover:bg-[#2a2838] rounded-[20px] border-[1px] border-white border-solid transition">
        <div>{children}</div>
      </div>
    </div>
  );

  if (isExternal) {
    return (
      <a href={href} target="_blank" rel="noreferrer noopener" {...props}>
        {Button}
      </a>
    );
  }

  return (
    <Link href={href}>
      <a {...props}>{Button}</a>
    </Link>
  );
};

export default LinkButton;
