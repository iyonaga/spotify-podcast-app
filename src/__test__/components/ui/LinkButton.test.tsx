import { render, screen } from '@testing-library/react';
import LinkButton from '@/components/ui/LinkButton';

const href = 'https://example.com';
const props = {
  href,
};

describe('components/ui/LinkButton.tsx', () => {
  test('スナップショットが一致する', () => {
    const { asFragment, rerender } = render(
      <LinkButton {...props}>link text</LinkButton>
    );

    expect(asFragment()).toMatchSnapshot();
    rerender(
      <LinkButton {...props} isExternal={true}>
        link text
      </LinkButton>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('外部リンクは新しいタブで開く', () => {
    render(
      <LinkButton {...props} isExternal={true}>
        link text
      </LinkButton>
    );
    const link = screen.getByRole('link');
    expect(link).toHaveAttribute('href', href);
    expect(link).toHaveAttribute('target', '_blank');
    expect(link).toHaveAttribute('rel', 'noreferrer noopener');
  });
});
