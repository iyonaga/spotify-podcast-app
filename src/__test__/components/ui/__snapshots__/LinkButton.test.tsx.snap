// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ui/LinkButton.tsx スナップショットが一致する 1`] = `
<DocumentFragment>
  <a
    href="https://example.com"
  >
    <div
      class="inline-block"
    >
      <div
        class="flex justify-center items-center px-[20px] min-w-[160px] h-[40px] text-14px font-bold hover:bg-[#2a2838] rounded-[20px] border-[1px] border-white border-solid transition"
      >
        <div>
          link text
        </div>
      </div>
    </div>
  </a>
</DocumentFragment>
`;

exports[`components/ui/LinkButton.tsx スナップショットが一致する 2`] = `
<DocumentFragment>
  <a
    href="https://example.com"
    rel="noreferrer noopener"
    target="_blank"
  >
    <div
      class="inline-block"
    >
      <div
        class="flex justify-center items-center px-[20px] min-w-[160px] h-[40px] text-14px font-bold hover:bg-[#2a2838] rounded-[20px] border-[1px] border-white border-solid transition"
      >
        <div>
          link text
        </div>
      </div>
    </div>
  </a>
</DocumentFragment>
`;
